serial/serial_resources/component_instance.py

```diff
@@ -0,0 +1,112 @@
+import serial
+from ..api_client import APIClient
+from .component_instance_link import ComponentInstanceLink
+
+class ComponentInstance:
+    def __init__(self, component_instance_data):
+        self.client = APIClient(serial.api_key, serial.base_url)
+        self.data = component_instance_data
```

Comment:
I did several of these, where the object type stores the api result in a data field - it works but it's not very elegant. I think ideally, it would deconstruct the data into its own properties automatical/automagically. Like you'd do the load, and automatically have a .id, .componet_id property, etc.

Reply:
I think this is reasonable for now. It's much simpler this way and once we understand the usage patterns of the library we can make improvements